Shadow 9 (beta13) throws exception if a .so file is included in the runtime classpath

### Expected and Results

Adding a dependency on something like `io.github.ganadist.sqlite4java:libsqlite4java-osx-aarch64:1.0.392` in the implementation classpath of a project causes binary libs (.dylib, .so files) to get added to the runtime classpath. When the shadow plugin (v9) encounters these, it tries to treat them as zip files and fails:

```
Caused by: java.util.zip.ZipException: Archive is not a ZIP archive	
at org.apache.commons.compress.archivers.zip.ZipFile.positionAtEndOfCentralDirectoryRecord(ZipFile.java:562)	
•••
at com.github.jengelman.gradle.plugins.shadow.tasks.ShadowCopyAction.execute(ShadowCopyAction.kt:54)	
•••
at com.github.jengelman.gradle.plugins.shadow.tasks.ShadowJar.copy(ShadowJar.kt:316)
```

Version 8.3.6 of the shadow plugin handled this without throwing this exception.

### Related environment and versions

Using gradle 8.14.1, shadow gradle plugin 9.0.0-beta13, on macOS 15.5 (24F74)

### Reproduction steps

Put this in a build.gradle.kts file:
```
buildscript {
  repositories {
    mavenCentral()
  }
}
repositories {
   mavenCentral()
}

plugins {
  id("org.jetbrains.kotlin.jvm") version ("2.1.21")
  id("com.gradleup.shadow") version ("9.0.0-beta13")
}

dependencies {
  implementation("io.github.ganadist.sqlite4java:libsqlite4java-osx-aarch64:1.0.392")
}
```

Run `gradle shadowJar`

### Anything else?

_No response_

--- a/docs/changes/README.md
+++ b/docs/changes/README.md
@@ -142,8 +142,8 @@ Options:
   `com.github.jengelman.gradle.plugins.shadow.tasks`. ([#1272](https://github.com/GradleUp/shadow/pull/1272))
 - **BREAKING CHANGE:** Align the behavior of `ShadowTask.from` with Gradle's `AbstractCopyTask.from`. ([#1233](https://github.com/GradleUp/shadow/pull/1233))  
   In the previous versions, `ShadowTask.from` would always unzip the files before processing them, which caused serial
-  issues that are hard to fix. Now it behaves like Gradle's `AbstractCopyTask.from`, which means it will not unzip the files, only
-  copy the files as-is. If you still want to shadow the unzipped files, try out something like:
+  issues that are hard to fix. Now it behaves like Gradle's `AbstractCopyTask.from`, which means it will not unzip
+  the files, only copy the files as-is. If you still want to shadow the unzipped files, try out something like:
   ```kotlin
     tasks.shadowJar {
       from(zipTree(files('path/to/your/file.zip')))

